conans/client/generators/cmake_find_package.py

```diff
@@ -43,52 +48,256 @@ class CMakeFindPackageGenerator(Generator):
         endif()
         """)
 
+    find_components_tpl = Template(textwrap.dedent("""\
+        ########## MACROS ###########################################################################
+        #############################################################################################
+        {{ conan_message }}
+        {{ conan_find_apple_frameworks }}
+        {{ conan_package_library_targets }}
+
+        ########### FOUND PACKAGE ###################################################################
+        #############################################################################################
+
+        include(FindPackageHandleStandardArgs)
+
+        conan_message(STATUS "Conan: Using autogenerated Find{{ pkg_name }}.cmake")
+        set({{ pkg_name }}_FOUND 1)
+        set({{ pkg_name }}_VERSION "{{ pkg_version }}")
+        set({{ pkg_name }}_COMPONENTS {{ pkg_components }})
+
+        find_package_handle_standard_args({{ pkg_name }} REQUIRED_VARS
+                                          {{ pkg_name }}_VERSION VERSION_VAR {{ pkg_name }}_VERSION)
+        mark_as_advanced({{ pkg_name }}_FOUND {{ pkg_name }}_VERSION)
+
+        if({{ pkg_name }}_FIND_COMPONENTS)
+            foreach(_FIND_COMPONENT {{ '${'+pkg_name+'_FIND_COMPONENTS}' }})
+                list(FIND {{ pkg_name }}_COMPONENTS "{{ pkg_name }}::${_FIND_COMPONENT}" _index)
+                if(${_index} EQUAL -1)
```

Comment:
Do we want to consider here the inputs `QUIET` and `REQUIRED`?

Reply:
Aren't those variables managed by CMake itself? From https://cmake.org/cmake/help/v3.0/command/find_package.html

> The QUIET option disables messages if the package cannot be found.
> The MODULE option disables the second signature documented below.
> The REQUIRED option stops processing with an error message if the package cannot be found.

Reply:
Oh! Maybe it uses the variable `<PKG>_FOUND`?